text/0894-eslint-plugin-import.md

```diff
@@ -0,0 +1,174 @@
+---
+stage: accepted
+start-date: 2023-01-07T00:00:00.000Z
+release-date: # In format YYYY-MM-DDT00:00:00.000Z
+release-versions:
+teams: # delete teams that aren't relevant
+  - cli
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/894
+project-link:
+suite:
+---
+
+<!---
+Directions for above:
+
+stage: Leave as is
+start-date: Fill in with today's date, 2032-12-01T00:00:00.000Z
+release-date: Leave as is
+release-versions: Leave as is
+teams: Include only the [team(s)](README.md#relevant-teams) for which this RFC applies
+prs:
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+suite: Leave as is
+-->
+
+# Add eslint-plugin-import to ember-cli blueprint
```

Comment:
I have a config for no-extraneous-imports, which we need, but i don't know if plugin-import supports it.

Why plugin-import over plugin-n?

Reply:
What do you mean exactly? The ember-cli blueprint already includes [eslint-plugin-node](https://github.com/mysticatea/eslint-plugin-node). [eslint-plugin-n](https://github.com/eslint-community/eslint-plugin-n) is intended to be a temporary fork of eslint-plugin-node, until the eslint-community team is able to take over eslint-plugin-node.

Reply:
eslint-plugin-node is unmaintained and doesn't support ESM or exports.

eslint-plugin-n has rules (such as no-extraneous-import), that work well within browser code -- because every imported module _must_ be declared in the package.json -- this rule is very good

Reply:
Replacing eslint-plugin-node with eslint-plugin-n, at least temporarily, is worth considering given that the owner of eslint-plugin-node has disappeared. But eslint-plugin-import has many other distinct rules beyond these Node plugins. Are you suggesting eslint-plugin-import isn't worthwhile over the existing linting?

Reply:
Looking over this list:
```
plugin:import/recommended
```
https://github.com/import-js/eslint-plugin-import -- config here: https://github.com/import-js/eslint-plugin-import/blob/main/config/recommended.js#L8

this does actually still look useful, though the set of recommended rules seems actually kinda small.

we should do a separate RFC for eslint-plugin-n (dropping eslint-plugin-node, and adding additional rules for browser code)

Disregard my original comment!

This RFC still brings up a ton of other valid stuff that we should def implement.
Incremental improvements is key! (no need to have perfection out of the gate -- though, I think it's weird that eslint-plugin-import still has no analogous rule to no-extraneous-import :upside_down_face: )

Reply:
FYI https://github.com/ember-cli/ember-cli/pull/10060

Reply:
I'm outdated!